parser/src/main/java/com/microsoft/gctoolkit/parser/jvm/UnifiedJVMConfiguration.java

```diff
@@ -191,22 +200,25 @@ private void discoverCollector(String line) {
         }
     }
 
+    /**
+     *
+     * @param line
+     */
     private void discoverDetails(String line) {
 
-        if (collectionCount > 2)
+        if (collectionCount > CYCLES_TO_EXAMINE_BEFORE_GIVING_UP)
             getDiary().setFalse(ADAPTIVE_SIZING, TLAB_DATA, PRINT_REFERENCE_GC, PRINT_PROMOTION_FAILURE, PRINT_FLS_STATISTICS, PRINT_HEAP_AT_GC);
 
         if (CYCLE_START.parse(line) != null) {
             getDiary().setTrue(SupportedFlags.GC_CAUSE);
-            collectionCount++;
         }
     }
 
     private void discoverJVMEvents(String line) {
         GCLogTrace trace = GC_COUNT.parse(line);
         if (trace == null) return;
         int gcCurrentEventCount = trace.getIntegerGroup(1);
-        if (gcCurrentEventCount > 2) {
+        if (gcCurrentEventCount > CYCLES_TO_EXAMINE_FOR_SAFEPOINT) {
```

Comment:
Won't this fail if you are in a rollover log? If I understand this (poorly named method) correctly, if you don't see 'safepoint' before you see GC(3), then these flags are set to false. But couldn't you have a log that starts at GC(1234) and still has 'safepoint' entries?

Reply:
A complete GC cycle can't happen without a safepoint.

Reply:
Right, but this isn't counting cycles, it's just looking at the GC id. Or am I missing something?